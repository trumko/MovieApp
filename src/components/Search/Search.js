import React, { Component } from "react";
import { withRouter } from 'react-router-dom'
import classnames from 'classnames'

import './Search.scss';

export class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {
      type: 'title',
      inputQuery: '',
    };
  }

  setSearchByType = (type) => {
    this.setState({ type: type });
  }

  onInputChange = (e) => {
    this.setState({
      inputQuery: e.currentTarget.value
    });
  }

  onFormSubmit = (e) => {
    e.preventDefault();
    if (this.state.inputQuery) {
      console.log('this.props.history', this.props.history);
      this.props.history.push(`/search?search=${this.state.inputQuery}&genre=${this.state.type}`);
    }
  }

  render() {
    return (
      <div className='mr_search'>
        <h1>Find your movie</h1>
        <form onSubmit={this.onFormSubmit}>
          <input
            className="mr_searchInput"
            type="text"
            name="search-query"
            placeholder="search..."
            value={this.state.inputQuery}
            onChange={this.onInputChange}
          />
          <hr />
          <button className="mr_searchSubmit" type="submit" name="search">search</button>
          <div className="mr_searchSwitch">
            <span>Search by</span>
            <button
              type="button"
              className={classnames({'mr_isActiveType': this.state.type === 'title'})}
              onClick={() => this.setSearchByType('title')}
              data-search-type="title"
            >
              title
            </button>
            <button
              type="button"
              className={classnames({'mr_isActiveType': this.state.type === 'genres'})}
              onClick={() => this.setSearchByType('genres')}
              data-search-type="genres"
            >
              genre
            </button>
          </div>
        </form>
        <div className="mr_subHeader" />
      </div>
    )
  }
}

export default withRouter(Search)
