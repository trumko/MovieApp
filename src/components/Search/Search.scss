@import "styles/variables.scss";

.mr_search {
  color: $white;

  form {
    position: relative;
    height: 80px;
    margin-bottom: 20px;
  }

  button {
    color: $white;
    background: $pink;
    border: none;
    text-transform: uppercase;
    font-weight: bold;
    cursor: pointer;

    &:hover {
      background-color: $pinkLight;
    }
  }

  h1 {
    text-transform: uppercase;
    font-size: 14px;
    margin: 0px 0 20px 0;
  }

  .mr_searchInput {
    width: 100%;
    background-color: $black;
    border: none;
    height: 40px;
    padding: 0 10px;
    color: $white;
  }

  .mr_searchSubmit {
    position: absolute;
    right: 0;
    bottom: 0;
    padding: 7px 50px;
    font-size: 12px;
  }

  hr {
    border-color: $pink;
    margin-top: 0;
  }

  .mr_searchSwitch {
    display: inline-block;
    margin-top: 8px;

    span {
      font-size: 10px;
      text-transform: uppercase;
      font-weight: bold;
      padding-right: 15px;
    }

    button {
      font-size: 10px;
      margin-right: 15px;
      padding: 5px 10px;

      &.mr_isActiveType {
        background: $greyDark;
      }
    }
  }
}
