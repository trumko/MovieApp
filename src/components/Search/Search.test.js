import React from "react"
// import { create } from "react-test-renderer"
import { shallow } from 'enzyme'
import { Search } from "./Search"

describe("Search", () => {
  test("should input field work correctly", () => {
    const component = shallow(<Search />);
    component.find('.mr_searchInput').simulate('change', {currentTarget: {value: 'This is movie title'}});
    expect(component.state('inputQuery')).toBe('This is movie title');
    expect(component).toMatchSnapshot();
  });

  test("should switch search types", () => {
    const component = shallow(<Search />);
    component.find('.mr_searchSwitch button[data-search-type="genres"]').simulate('click');
    expect(component.state().type).toBe("genres");
    component.find('.mr_searchSwitch button[data-search-type="title"]').simulate('click');
    expect(component.state().type).toBe("title");
  });
});

// import React, { Component } from "react";
// import { withRouter } from 'react-router-dom'
// import classnames from 'classnames'
//
// import './Search.scss';
//
// export class Search extends Component {
//   constructor(props) {
//     super(props);
//     this.state = {
//       type: 'title',
//       inputQuery: '',
//     };
//   }
//
//   setSearchByType = (type) => {
//     this.setState({ type: type });
//   }
//
//   onInputChange = (e) => {
//     this.setState({
//       inputQuery: e.currentTarget.value
//     });
//   }
//
//   onFormSubmit = (e) => {
//     e.preventDefault();
//     if (this.state.inputQuery) {
//       this.props.history.push(`/search?search=${this.state.inputQuery}&genre=${this.state.type}`);
//     }
//   }
//
//   render() {
//     return (
//       <div className='mr_search'>
//         <h1>Find your movie</h1>
//         <form onSubmit={this.onFormSubmit}>
//           <input
//             className="mr_searchInput"
//             type="text"
//             name="search-query"
//             placeholder="search..."
//             value={this.state.inputQuery}
//             onChange={this.onInputChange}
//           />
//           <hr />
//           <button className="mr_searchSubmit" type="submit" name="search">search</button>
//           <div className="mr_searchSwitch">
//             <span>Search by</span>
//             <button
//               type="button"
//               className={classnames({'mr_isActiveType': this.state.type === 'title'})}
//               onClick={() => this.setSearchByType('title')}
//             >
//               title
//             </button>
//             <button
//               type="button"
//               className={classnames({'mr_isActiveType': this.state.type === 'genres'})}
//               onClick={() => this.setSearchByType('genres')}
//             >
//               genre
//             </button>
//           </div>
//         </form>
//       </div>
//     )
//   }
// }
//
// export default withRouter(Search)
