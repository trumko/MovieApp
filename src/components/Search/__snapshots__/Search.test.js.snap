// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should input field work correctly 1`] = `
<div
  className="mr_search"
>
  <h1>
    Find your movie
  </h1>
  <form
    onSubmit={[Function]}
  >
    <input
      className="mr_searchInput"
      name="search-query"
      onChange={[Function]}
      placeholder="search..."
      type="text"
      value="This is movie title"
    />
    <hr />
    <button
      className="mr_searchSubmit"
      name="search"
      type="submit"
    >
      search
    </button>
    <div
      className="mr_searchSwitch"
    >
      <span>
        Search by
      </span>
      <button
        className="mr_isActiveType"
        data-search-type="title"
        onClick={[Function]}
        type="button"
      >
        title
      </button>
      <button
        className=""
        data-search-type="genres"
        onClick={[Function]}
        type="button"
      >
        genre
      </button>
    </div>
  </form>
</div>
`;
